pkg/rbuser/rbUser.go

```diff
@@ -0,0 +1,74 @@
+package rbuser
+
+import (
+	"fmt"
+	"html/template"
+	"os"
+	"time"
+)
+
+// RBUser is the redbric ldap user
+type RBUser struct {
+	UID              string
+	UserType         string
+	ObjectClass      string
+	Newbie           bool   // New this year
+	CN               string // Full name
+	Altmail          string // Alternate email
+	ID               int    // DCU ID number
+	Course           string // DCU course code
+	Year             int    // DCU course year number/code
+	YearsPaid        int    // Number of years paid (integer)
+	Updatedby        string // Username of user last to update
+	Updated          time.Time
+	CreatedBy        string // Username of user that created them
+	Created          time.Time
+	Birthday         time.Time
+	UIDNumber        int
+	GidNumber        int
+	Gecos            string
+	LoginShell       string
+	HomeDirectory    string
+	UserPassword     string   // Crypted password.
+	Host             []string // List of hosts.
+	ShadowLastChange time.Time
+}
+
+// Vhost reutrn apache  macro template
+func (u *RBUser) Vhost() string {
+	initial := []rune(u.UID)[0]
+	return fmt.Sprintf("use VHost /storage/webtree/%s/%s %s %s %s", string(initial), u.UID, u.UID, u.ObjectClass, u.UID)
+}
+
+// PrettyPrint output user info to command line
+func (u *RBUser) PrettyPrint() error {
+	const output = `
+	User Information
+	================
+	{{ with .UID }}uid: {{ . }}
```

Comment:
Wouldn't `uid: {{ .UID }}` not be better? What's the reason to use `with`?

Reply:
`with` is basically an if statement it means it only prints if the variable exists